hotshot-query-service/src/data_source/storage/sql/transaction.rs

```diff
@@ -399,17 +401,38 @@ impl Transaction<Write> {
         }
         query_builder.push(format!(" ON CONFLICT ({pk}) DO UPDATE SET {set_columns}"));
 
-        let res = self.execute(query_builder.build()).await?;
-        let stmt = query_builder.sql();
-        let rows_modified = res.rows_affected() as usize;
-
-        if rows_modified != num_rows {
-            tracing::error!(
-                stmt,
-                "unexpected number of rows modified: expected {num_rows} but got {rows_modified}"
-            );
+        let interval = Duration::from_secs(1);
+        let mut retries = 5;
+
+        loop {
+            let query = query_builder.build();
+            let statement = query.sql();
+            match self.execute(query).await {
+                Ok(res) => {
+                    let rows_modified = res.rows_affected() as usize;
+                    if rows_modified as usize != num_rows {
+                        tracing::error!(
```

Comment:
Do you know when this would happen? Currently we log an error but don't return an error. Is it something we should investigate? We could leave some more information in the error message.